src/libs/chatAttachmentTokenHeaders.js

```diff
@@ -0,0 +1,21 @@
+import lodashGet from 'lodash/get';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import CONST from '../CONST';
+
+let encryptedAuthToken = '';
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: session => encryptedAuthToken = lodashGet(session, 'encryptedAuthToken', ''),
+});
```

Comment:
@Beamanator 
This is not the "React Way" to deal with global or storage state 
It's a bit of an antipattern actually because it doesn't integrate with component lifecycle - only a re-render would cause users of `chatAttachmentTokenHeaders` to re-evaluate, but the change in `ONYXKEYS.SESSION` would not trigger a re-render, unless the component is also subscribed to session 

The "React Way" is to capture this with `withOnyx` so the component is subscribed to session and re-evaluates headers the moment they change
To keep it simple the `withOnyx` can be applied on the core `Image` (FastImage) component and let it perform the logic from this file

I know we use the static `Onyx.connect` logic for other cases, but we should distinct the cases that really need it and the cases that can use `withOnyx` instead 
Another bad thing about the static pattern is that it is static - we can't lazy load it and the connection stays forever, while using the "React Way" would free resources
- the connection can be initiated only when needed
-  the connection is released when the component is no longer mounted

Reply:
Gooooood points, I overlooked those lifecycle points & didn't think about lazy loading / freeing resources, I'll definitely keep those in mind in the future! Next round will not do this this way ;D